src/csharp/Tokenizer.cs

```diff
@@ -0,0 +1,133 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Runtime.InteropServices;
+using System.Text;
+
+namespace Microsoft.ML.OnnxRuntimeGenAI
+{
+    public class Tokenizer : IDisposable
+    {
+        private IntPtr _tokenizerHandle;
+        private bool _disposed = false;
+
+        public Tokenizer(Model model)
+        {
+            Result.VerifySuccess(NativeMethods.OgaCreateTokenizer(model.Handle, out _tokenizerHandle));
+        }
+
+        public Sequences EncodeBatch(string[] strings)
+        {
+            IntPtr stringArray = IntPtr.Zero;
+            UIntPtr[] stringLengths = new UIntPtr[strings.Length];
+            for (int i = 0; i < strings.Length; i++)
+            {
+                stringLengths[i] = (UIntPtr)UTF8Encoding.UTF8.GetByteCount(strings[i]);
+            }
+
+            unsafe
+            {
+                fixed (UIntPtr* stringLengthsPtr = stringLengths)
```

Comment:
fixed is not necessary here. UIntPtr is a blittable type. Change the function signature to accept that and it will do the pinning for you.